server/src/main/java/org/elasticsearch/action/support/SubscribableListener.java

```diff
@@ -191,6 +191,10 @@ public final void addListener(ActionListener<T> listener) {
      *                      <li>Pass the desired executor in as {@code executor}, and</li>
      *                      <li>Invoke {@link #addListener} using that executor.</li>
      *                      </ul>
+     *                      <p>
+     *                      If {@code executor} rejects the execution of the completion of the subscribing listener then the result is
+     *                      discarded and the subscribing listener is completed with a rejection exception on the thread which completes
```

Comment:
```suggestion
     *                      discarded and the subscribing listener is completed with an execution exception on the thread which completes
```

Reply:
I think I'd interpret that as meaning a subclass of `java.util.concurrent.ExecutionException` which isn't the case for `RejectedExecutionException` or `EsRejectedExecutionException` - the docs for `ExecutionException` say it is "thrown when attempting to retrieve the result of a task that aborted by throwing an exception" which doesn't apply here.

(acknowledging that `RejectedExecutionException` does have the suffix `ExecutionException` but I'd rather call it a rejection exception in this context)